components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -40,11 +37,15 @@ export class DesktopDetailComponent implements OnInit {
     unknown: 'help',
     missing: 'help'
   };
+  private subscription: Subscription;
+  private isDestroyed: Subject<boolean> = new Subject<boolean>();
 
   constructor(
     private store: Store<NgrxStateAtom>
   ) {
-    this.store.select(getDailyNodeRuns).subscribe((dailyNodeRuns) => {
+    this.subscription = this.store.select(getDailyNodeRuns).pipe(
```

Comment:
Using the `OnDestroy` lifecycle hook makes subscription management automatic--no need to do it manually:
```suggestion
    this.store.select(getDailyNodeRuns).pipe(
```